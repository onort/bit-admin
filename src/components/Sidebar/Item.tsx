import React from "react"
import { NavLink } from "react-router-dom"
import cx from "classnames"

import styles from "./Sidebar.scss"
import { AppRoute } from "../../routes"

interface Props {
  className?: string
  icon: React.ReactNode
  title: string
  to: AppRoute
}

const Item: React.SFC<Props> = props => {
  const { className, icon, title, to } = props
  const componentClassName = cx(styles.item, className)
  return (
    <NavLink
      to={to}
      className={componentClassName}
      activeClassName={styles.active}
    >
      <span className={styles.itemIcon}>{icon}</span>
      <span className={styles.itemTitle}>{title}</span>
    </NavLink>
  )
}

export default Item
