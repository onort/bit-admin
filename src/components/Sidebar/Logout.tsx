import React from "react"
import { MdExitToApp as LogoutIcon } from "react-icons/md"
import cx from "classnames"

import styles from "./Sidebar.scss"
import { AuthConsumer } from "../../context/"

const Logout: React.SFC = () => {
  const className = cx(styles.item, styles.logout)
  const handleLogout = (ctx: any): any => async () => {
    console.log("handleLogin in logout component ran")
    await ctx.onLogout()
  }

  return (
    <AuthConsumer>
      {(ctx: any) => (
        <a className={className} onClick={handleLogout(ctx)}>
          <span className={styles.itemIcon}>
            <LogoutIcon />
          </span>
          <span className={styles.itemTitle}>Logout</span>
        </a>
      )}
    </AuthConsumer>
  )
}

export default Logout
