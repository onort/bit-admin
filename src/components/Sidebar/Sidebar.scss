@import "../../styles/main.scss";

.container {
  position: fixed;
  top: $appBar-height;
  left: 0;
  width: $sidebar-width;
  padding: $spacer-lg;
  background-color: $dark-bg;
  color: $white;
  min-height: calc(100vh - #{$appBar-height});
  overflow-y: hidden;
  transition: all 0.3s;

  &.closed {
    left: -$sidebar-width;
    width: 0;
    padding: 0;
  }
}
