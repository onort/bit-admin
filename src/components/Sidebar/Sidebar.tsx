import React from "react"
import cx from "classnames"
import {
  MdAdd as AddIcon,
  MdViewList as ViewBitsIcon,
  MdViewStream as ViewTagsIcon,
  MdLocalOffer as TagIcon
} from "react-icons/md"

import styles from "./Sidebar.scss"
import { SidebarItem, SidebarLogout } from "./"

interface Props {
  className?: string
  sidebarOpen: boolean
}

const Sidebar: React.SFC<Props> = props => {
  const className = cx(
    styles.container,
    { [styles.closed]: !props.sidebarOpen },
    props.className
  )
  return (
    <nav className={className}>
      <SidebarItem icon={<AddIcon />} title="Add Bit" to="/add-bit" />
      <SidebarItem icon={<TagIcon />} title="Add Tag" to="/add-tag" />
      <SidebarItem icon={<ViewBitsIcon />} title="View Bits" to="/view-bits" />
      <SidebarItem icon={<ViewTagsIcon />} title="View Tags" to="/view-tags" />
      <SidebarLogout />
    </nav>
  )
}

export default Sidebar
