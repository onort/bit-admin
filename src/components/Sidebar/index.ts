import Sidebar from "./Sidebar"
import SidebarItem from "./Item"
import SidebarLogout from "./Logout"

export default Sidebar

export { SidebarItem, SidebarLogout }
