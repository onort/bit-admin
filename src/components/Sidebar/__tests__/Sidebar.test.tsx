import React from "react"
import { mount, shallow } from "enzyme"
import { MemoryRouter as Router, NavLink } from "react-router-dom"

import Sidebar from "../"

describe("<Sidebar />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Router>
        <Sidebar sidebarOpen={true} />
      </Router>
    )
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have correct amount of links", () => {
    const wrapper = mount(
      <Router>
        <Sidebar sidebarOpen={true} />
      </Router>
    )
    expect(wrapper.find(NavLink)).toHaveLength(4)
    wrapper.unmount()
  })
})
