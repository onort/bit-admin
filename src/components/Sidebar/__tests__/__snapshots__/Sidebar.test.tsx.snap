// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should match snapshot 1`] = `"<nav class=\\"container\\"><a class=\\"item\\" href=\\"/add-bit\\"><span class=\\"itemIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path></svg></span><span class=\\"itemTitle\\">Add Bit</span></a><a class=\\"item\\" href=\\"/add-tag\\"><span class=\\"itemIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M21.41 11.58l-9-9C12.05 2.22 11.55 2 11 2H4c-1.1 0-2 .9-2 2v7c0 .55.22 1.05.59 1.42l9 9c.36.36.86.58 1.41.58.55 0 1.05-.22 1.41-.59l7-7c.37-.36.59-.86.59-1.41 0-.55-.23-1.06-.59-1.42zM5.5 7C4.67 7 4 6.33 4 5.5S4.67 4 5.5 4 7 4.67 7 5.5 6.33 7 5.5 7z\\"></path></svg></span><span class=\\"itemTitle\\">Add Tag</span></a><a class=\\"item\\" href=\\"/view-bits\\"><span class=\\"itemIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M4 14h4v-4H4v4zm0 5h4v-4H4v4zM4 9h4V5H4v4zm5 5h12v-4H9v4zm0 5h12v-4H9v4zM9 5v4h12V5H9z\\"></path></svg></span><span class=\\"itemTitle\\">View Bits</span></a><a class=\\"item\\" href=\\"/view-tags\\"><span class=\\"itemIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M4 18h17v-6H4v6zM4 5v6h17V5H4z\\"></path></svg></span><span class=\\"itemTitle\\">View Tags</span></a><a class=\\"item logout\\"><span class=\\"itemIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"></path></svg></span><span class=\\"itemTitle\\">Logout</span></a></nav>"`;
